src/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,54 @@
+import { JSX } from '@redneckz/uni-jsx';
+import type { JSXBlock, UniBlockProps } from '../../model/JSXBlock';
+import type { CarouselContent } from './CarouselContent';
+import { useState } from '@redneckz/uni-jsx/lib/hooks';
+import { renderNavButtons } from './renderNavButtons';
+import { BlockWrapper } from '../../ui-kit/BlockWrapper';
+import { CarouselItemsList } from './CarouselItemsList';
+
+export interface CarouselProps extends CarouselContent, UniBlockProps {}
+
+export const Carousel: JSXBlock<CarouselProps> = JSX<CarouselProps>(
+  ({
+    className = '',
+    context,
+    galleryBlocks = [],
+    cardShift = 0,
+    isBlur = true,
+    isCycle = false,
+    cardFullViewCount = 1,
+    version,
+    ...rest
+  }) => {
+    const [activeCardIndex, setActiveCardIndex] = useState(0);
+    const GalleryBlurClasses = isBlur ? 'absolute top-0 bottom-0 w-[84px]' : '';
+    const CardFullViewStyles =
+      cardFullViewCount === 1
+        ? `translateX(-${activeCardIndex * 100}%)`
+        : `translateX(-${activeCardIndex * cardShift}px)`;
+
+    return (
+      <BlockWrapper context={context} className={`box-border min-h-[100%] ${className}`} {...rest}>
+        <div
+          className="flex min-w-[100%] min-h-[100%] bg-white duration-1000 gap-9"
+          style={{ transform: `translateY(100%) ${CardFullViewStyles}` }}
+          role="list"
+        >
+          <CarouselItemsList>{galleryBlocks}</CarouselItemsList>
```

Comment:
А действительно ли надо было было вот этот вот <ul> со сбросом стилей выносить в отдельный компонент?
```suggestion
          <ul className="list-none m-0 p-0">{galleryBlocks}</ul>
```

Компонент CarouselItemsList только тут и используется. А если всё-таки хочется вынести, то надо выносить в ui-kit и заменять на этот wrapper везде, где используется такой же <ul>